packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -189,42 +200,55 @@ void YogaLayoutableShadowNode::appendYogaChild(
       !getTraits().check(ShadowNodeTraits::Trait::LeafYogaNode));
 
   ensureYogaChildrenLookFine();
-
-  yogaLayoutableChildren_.push_back(childNode);
-  yogaNode_.insertChild(&childNode->yogaNode_, yogaNode_.getChildren().size());
+    
+  runForEveryConcreteSubtree(childNode, [&](const YogaLayoutableShadowNode::Shared subtreeRoot) {
+    yogaLayoutableChildren_.push_back({
+      .directChild = childNode,
+      .yogaChild = subtreeRoot,
+    });
+    yogaNode_.insertChild(&subtreeRoot->yogaNode_, yogaNode_.getChildren().size());
+  });
 
   ensureYogaChildrenLookFine();
 }
 
 void YogaLayoutableShadowNode::adoptYogaChild(size_t index) {
+  if (hasDisplayContentsStyle()) {
+    return;
+  }
```

Comment:
It feels like a logic error if we try to adopt here. Could we assert instead?

Reply:
Also a nit, we should add this after the preconditions below

Reply:
Done in https://github.com/facebook/react-native/pull/46584/commits/bc1984f5c3ddc2a529c842377a2dbecb7c679dab and https://github.com/facebook/react-native/pull/46584/commits/dd283e0653a7a3587e85b1ae482bcde7414b176b.

I think early return was a leftover from when not every cloning path was updated to consider `display: contents`